src/components/carbon-provider/top-modal-context.tsx

```diff
@@ -10,6 +10,9 @@ type TopModalContextProps = {
   topModal: HTMLElement | null;
 };
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const windowAsAny = window as any;
```

Comment:
```suggestion
let windowAsAny: {
  __CARBON_INTERNALS_MODAL_SETTER_LIST: React.Dispatch<
    React.SetStateAction<HTMLElement | null>
  >[];
};
```
Could this work here?